### PR TITLE
syncyomi: init at 1.1.1

### PR DESCRIPTION
**Pull request by eriedaberrie**
## Description of changes

[SyncYomi](https://github.com/SyncYomi/SyncYomi), an open-source project to synchronize Tachiyomi manga reading progress and library across multiple devices.

I've also written a NixOS service module that is working for me so far but will probably need some more polishing before a PR (documentation, hardening, adding postgres support, etc.).

This replaces my previous PR #301155 which needed to fix several build errors and then another upstream release happened in the meantime, so I opted to make a new PR for this (also because I had accidentally based that PR off of my fork's master branch which was slightly inconvenient for me).

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sy/syncyomi/package.nix [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}:
+
+buildGoModule rec {
+  pname = "syncyomi";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "SyncYomi";
+    repo = "SyncYomi";
+    rev = "refs/tags/v${version}";
```
Comment (RossComputerGuy) [inline pkgs/by-name/sy/syncyomi/package.nix:19]: You do not need to specify `refs/tags/v` for it to pull a tag.

Comment (eriedaberrie) [reply pkgs/by-name/sy/syncyomi/package.nix:19]: In my original PR I was advised to add it to be explicit.

**Comment by RatCornu** [comment]
I made a module for this package: I'll make a PR when this will be merged!

**Comment by eriedaberrie** [comment]
I updated the PR to use the new-ish pnpm.fetchDeps.

File: pkgs/by-name/sy/syncyomi/package.nix (hunk 1 above)
Comment (eclairevoyant) [reply pkgs/by-name/sy/syncyomi/package.nix:17]: @RossComputerGuy `fetchFromGitHub` does not figure anything, github itself does. and this avoids ambiguity when there are identically named tags and branches, for example.

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 304470` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>syncyomi</li>
  </ul>
</details>

**Pull request merged**